### PR TITLE
Feature/port nodes to game objects

### LINKED ISSUE #89: Port nodes to gameObjects and Components
Create GameObject class.
Create BaseComponent class.

Port Nodes logic to new clases

### PR DESCRIPTION
**Pull request by theepsi**
closes #89

### CODE REVIEW

File: Equinox/GameObject.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+std::vector<GameObject*> GameObject::GetChilds()
+{
+	return _childs;
+}
+
+void GameObject::AddChild(GameObject* child)
+{
+	if (child != nullptr)
+	{
+		_childs.push_back(child);
```
Comment (jowie94) [inline Equinox/GameObject.cpp]: Set the parent of the child and remember to disassociate the child with its previous parent

File: Equinox/GameObject.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	if (component != nullptr)
+	{
+		component->Parent = this;
+		_components.push_back(component);
+	}
+}
+
+void GameObject::DeleteComponentByName(std::string name)
+{
+	//TODO:
```
Comment (jowie94) [inline Equinox/GameObject.cpp]: Implement this function, it can be interesting to have it in the future

File: Equinox/GameObject.cpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+{
+}
+
+GameObject::~GameObject()
+{
+}
+
+void GameObject::SetParent(GameObject* new_parent)
+{
+	_parent = new_parent;
```
Comment (jowie94) [inline Equinox/GameObject.cpp]: Add this object to the children list of the parent and remove it from the child list of the previous one

File: Equinox/Level.cpp [hunk 4]
```diff
@@ -39,30 +56,53 @@ void Level::Draw()
 	drawNode(root);
 }
 
-Level::Node* Level::FindNode(const char* name)
+GameObject* Level::FindNode(const char* name)
```
Comment (jowie94) [inline Equinox/Level.cpp]: Rename to FindGameObject

File: Equinox/Level.cpp [hunk 5]
```diff
@@ -39,30 +56,53 @@ void Level::Draw()
...
 {
 	return nullptr;
 }
 
-void Level::LinkNode(Node* node, Node* destination)
+void Level::LinkNode(GameObject* node, GameObject* destination)
```
Comment (jowie94) [inline Equinox/Level.cpp]: Just add a call destination->AddChild(node)

**Review by jowie94 — CHANGES_REQUESTED** [review]

**Review by jowie94 — APPROVED** [review]

**Review by polvila — APPROVED** [review]

**Pull request merged**